### PR TITLE
Configured CI

### PR DESCRIPTION
**Pull request by K-ballo**
Configured Travis and Appveyor.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+    - env: GCC_VERSION=5 CXXFLAGS="-std=c++11"
+      os: linux
+      addons: &gcc5
+        apt:
+          packages:
+            - cmake
+            - cmake-data
+            - g++-5
+          sources:
+            - george-edison55-precise-backports
```
Comment (mpark) [inline .travis.yml]: Is this a `precise` specific thing...? I'm not sure what it's for exactly, but
we should be on `trusty` by default now: https://blog.travis-ci.com/2017-08-31-trusty-as-default-status

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE.md or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+language: cpp
+sudo: false
+
+matrix:
+  include:
+    # gcc-4.8
```
Comment (mpark) [inline .travis.yml:14]: MPark.Variant currently doesn't support GCC 4.8.
At least at the time, I decided it wasn't worth the effort.
Do you think it's worth the effort to support 4.8?

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    - env: GCC_VERSION=4.9 CXXFLAGS="-std=c++11"
+      os: linux
+      addons: &gcc49
+        apt:
+          packages:
+            - g++-4.9
+          sources:
+            - ubuntu-toolchain-r-test
+
+    - env: GCC_VERSION=4.9 CXXFLAGS="-std=c++1y"
```
Comment (mpark) [inline .travis.yml]: GCC 4.9 supports `-std=c++14`.

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+      os: linux
+      addons: *gcc49
+
+    # gcc-5
+    - env: GCC_VERSION=5 CXXFLAGS="-std=c++11"
+      os: linux
+      addons: &gcc5
+        apt:
+          packages:
+            - cmake
```
Comment (mpark) [inline .travis.yml]: Hm, the 4.8/4.9 matrices above didn't include `cmake` nor `cmake-data`...
Are these required here and below for some reason not applicable to the 4.8/4.9 matrices?

__UPDATE__: Looking at the Travis logs, it looks like this is unnecessary.
```
cmake is already the newest version (3.2.2-2~ubuntu14.04.1~ppa1).
cmake-data is already the newest version (3.2.2-2~ubuntu14.04.1~ppa1).
```

File: .travis.yml [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+            - g++-7
+          sources:
+            - george-edison55-precise-backports
+            - ubuntu-toolchain-r-test
+
+    - env: GCC_VERSION=7 CXXFLAGS="-std=c++14"
+      os: linux
+      addons: *gcc7
+
+    - env: GCC_VERSION=7 CXXFLAGS="-std=c++1z"
```
Comment (mpark) [inline .travis.yml]: GCC 7 suports `-std=c++17`.

File: .travis.yml [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+        apt:
+          packages:
+            - cmake
+            - cmake-data
+            - clang-3.9
+            - libstdc++-6-dev
+          sources:
+            - george-edison55-precise-backports
+            - ubuntu-toolchain-r-test
+            - llvm-toolchain-precise-3.9
```
Comment (mpark) [inline .travis.yml]: We can do `llvm-toolchain-trusty-3.9` here.

File: .travis.yml [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+      addons: *clang39
+
+    - env: CLANG_VERSION=3.9 CXXFLAGS="-std=c++1z"
+      os: linux
+      addons: *clang39
+
+    # clang-4
+    - env: CLANG_VERSION=4.0 CXXFLAGS="-std=c++11"
+      os: linux
+      dist: trusty
```
Comment (mpark) [inline .travis.yml]: We should be on `trusty` by default now: https://blog.travis-ci.com/2017-08-31-trusty-as-default-status

File: .travis.yml [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+      addons: *xcode91
+
+branches:
+  only:
+    - master
+
+before_install:
+  - if [ -n "$GCC_VERSION" ]; then export CXX="g++-${GCC_VERSION}" CC="gcc-${GCC_VERSION}"; fi
+  - if [ -n "$CLANG_VERSION" ]; then export CXX="clang++-${CLANG_VERSION}" CC="clang-${CLANG_VERSION}"; fi
+  - export CXXFLAGS="${CXXFLAGS} -Wall -Wextra -Werror"
```
Comment (mpark) [inline .travis.yml]: Do we also want `-pedantic`?

**Review by mpark — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (K-ballo) [reply .travis.yml:14]: I do, to the extent that it is reasonable. I'm ready to take responsibility for it.

File: .travis.yml (hunk 1 above)
Comment (K-ballo) [reply .travis.yml]: I'll try taking these out, it'd be nice if things just work nowadays

File: .travis.yml (hunk 8 above)
Comment (K-ballo) [reply .travis.yml]: We definitively do!

File: .travis.yml [hunk 9]
```diff
@@ -0,0 +1,305 @@
+# SLB
+#
+# Copyright Michael Park, 2017
+# Copyright Agustin Berge, 2017
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
```
Comment (mpark) [inline .travis.yml]: Let's do:
```
# Distributed under the Boost Software License, Version 1.0.
# (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
```

File: .appveyor.yml [hunk 10]
```diff
@@ -0,0 +1,89 @@
+# SLB
+#
+# Copyright Michael Park, 2017
+# Copyright Agustin Berge, 2017
+#
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
```
Comment (mpark) [inline .appveyor.yml]: Let's do:
```
# Distributed under the Boost Software License, Version 1.0.
# (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
```

**Review by mpark — APPROVED** [review]

**Pull request merged**